lib/streamlit/dataframe_util.py

```diff
@@ -366,6 +392,23 @@ def convert_anything_to_pandas_df(
             else _fix_column_naming(pd.DataFrame(data))
         )
 
+    if is_polars_dataframe(data):
+        data = data.clone() if ensure_copy else data
+        return data.to_pandas()
```

Comment:
I have juist realized and probably should have done earlier, but I find it odd that the function sounds like a boolean check `is_polars_dataframe` but actually performs a (expensive?) `to_pandas()` operation. Also, should it have a return type annotation?

Reply:
You might have missed the `if` here :) The actual `is_polars_dataframe ` function implementation just does a type check.